tests/command_generate_output.py

```diff
@@ -0,0 +1,120 @@
+import unittest
+
+import contextlib
+import filecmp
+import glob
+import os
+import subprocess
+import sys
+import tempfile
+
+
+@contextlib.contextmanager
+def chdir(path):
+    cwd = os.getcwd()
+    try:
+        os.chdir(path)
+        yield
+    finally:
+        for file in glob.glob('*/*.out'):
+            os.remove(file)
+        os.chdir(cwd)
+
+def prepare_files(input_files):
+    for f in input_files:
+        if os.path.dirname(f['path']):
+            os.makedirs(os.path.dirname(f['path']), exist_ok=True)
+        with open(f['path'], 'w') as fh:
+            fh.write(f['data'])
+        if f.get('executable', False):
+            os.chmod(f['path'], 0o755)
+
+class TestGenerateOutput(unittest.TestCase):
+
+    def snippet_call_test(self, args, input_files, expected_values):
+        ojtools = os.path.abspath('oj')
+        with tempfile.TemporaryDirectory() as tempdir:
+            with chdir(tempdir):
+                prepare_files(input_files)
+                _ = subprocess.check_output([ojtools, 'generate-output'] + args, stderr=sys.stderr)
+                for expect in expected_values:
+                    with open(expect['path']) as f:
+                        self.assertEqual(''.join(f.readlines()), expect['data'])
+
+    def test_call_generate_output_simple(self):
+        self.snippet_call_test(
+            args=[ '-c', 'cat' ],
+            input_files=[
+                { 'path': 'test/sample-1.in', 'data': 'foo\n' },
+                { 'path': 'test/sample-2.in', 'data': 'bar\n' },
+            ],
```

Comment:
```suggestion
                { 'path': 'test/sample-2.out', 'data': 'baz\n' },
            ],
```

「既に出力側のファイルが存在する場合は無視する」という仕様もあるので、これのテストも必要です。